flask_app/blueprints/rest.py

```diff
@@ -97,13 +102,23 @@ def _get_iterator(self):
         else:
             returned = super(TestResource, self)._get_iterator()
 
+
+        #get session
         if args.session_id is not None:
+            using_logical_id = False
             try:
                 session_id = int(args.session_id)
             except ValueError:
-                returned = Test.query.join(Session).filter(Session.logical_id == args.session_id)
+                using_logical_id = True
+
+            session = Session.query.filter(Session.logical_id == args.session_id).first() if using_logical_id else \
+                      Session.query.filter(Session.id == session_id).first()
+            if session and session.children:
+                children_ids = [child.id for child in session.children]
```

Comment:
It's better to only fetch the ids for the children, and avoid all the computed fields - see lazily loaded fields in SQLAlchemy